Frontend supports only 18 decimal tokens

mariano-aguero opened the pull request:
Closes #516 

### Includes:

- Pass base token's decimals to `buy_sell.tsx`'s BigNumberInput, so the `amount` obtained is correctly expressed in base units.
- Fix `makerAssetAmount` calculation in the function `buildLimitOrder` from `src/util/orders.ts` (should be expressed in the taker asset's base units).
- Fix calculation of `price` in `buildLimitOrder` from `src/util/ui_orders.ts`.
- Adapt UI components that didn't considered decimals other than 18 to show a price (e.g. the markets dropddown).
- Fix failing tests due to changes in `src/util/orders.ts`.

Commit:
Fix to use base token decimals

Commit:
Add decimals from base currency pair

Commit:
Merge branch 'development' into hotfix/support-custom-decimals-tokens

Commit:
Refactor related to the networkId

unjapones commented:
Hey Mariano, I tested this yesterday and found out that there might be a problem withe the units handling. 
I created a `Sell` limit order with the following data:
- base token: `DGD, 9 decimals`
- quote token: `WETH, 18 decimals`
- value in "Amount" input: `2`
- value in "Price per token input: `1.5`

The above generated the following when MetaMask asked me to sign the order (screenshot):
![wrong_signed_order_data](https://user-images.githubusercontent.com/4421917/59269784-3659a000-8c26-11e9-871e-3f217c2a9c10.png)

I believe there is a problem around the amounts and that the expected `SignedOrder` should end up with the following values:
```
makerAssetAmount: 2000000000,
takerAssetAmount: 3000000000000000000
```
`makerAssetAmount` should be expressed in base units of the maker asset, and the same goes for  `takerAssetAmount` (which has more zeros because WETH has more decimals).

Please confirm that you see the same thing. We'll continue improving this PR's branch.

unjapones committed:
Remove duplicated submit code in buy_sell.tsx.

unjapones committed:
Fix util/order buildLimitOrder.

Fix util/order buildLimitOrder, maker/taker asset amount generation for
the SignedOrder not being in base units of their corresponding asset.

unjapones committed:
Fix ui_orders ordersToUIOrders price math.

unjapones committed:
Fix buildLimitOrders amount calculation on Buy.

Fix services/relayer market price calculation.

unjapones committed:
Fix takerAssetAmount calculation in limit.

Add getTokenByAssetData to known_tokens.

unjapones committed:
Fix ui_orders.test.ts by using actual KnownTokens.

unjapones committed:
Fix util/orders.test.ts.

mariano-aguero commented:
LGTM @unjapones , I think is ready to merge!
